packages/hadron-app-registry/src/register-plugin.tsx

```diff
@@ -194,6 +225,36 @@ export function registerHadronPlugin<
     },
     {
       displayName: config.name,
+      withMockServices(
+        mocks: Partial<Registries & Services<S>>
+      ): React.FunctionComponent<T> {
+        const {
+          globalAppRegistry: _globalAppRegistry,
+          localAppRegistry: _localAppRegistry,
+          ...mockServices
+        } = mocks;
+        const mockServiceLocators = Object.fromEntries(
+          Object.keys(services ?? {}).map((s) => {
+            return [s, mockServices[s] ? () => mockServices[s] : services?.[s]];
+          })
+        ) as unknown as S;
+        const MockPlugin = registerHadronPlugin(config, mockServiceLocators);
```

Comment:
Won’t it be an issue that this re-uses the same `registryName` as the original plugin and therefore shares a store (which may have already been initialized with the original services)?

Reply:
Hmmm, I think for that to conflict you'd need to actually render / activate both mocked and unmocked plugin in the same test, which sounds like a rare corner case, unless I'm missing some scenario here 🤔 Doesn't hurt to  override config name here though, generally not that we should rely on it in any way externally, can do that!

Reply:
Just needs to happen in the same process as long as the global app registry is truly global, right? That doesn’t seem all _that_ unlikely, and it’s going to be hard to diagnose because it could involve some “spooky action at a distance” from a completely unrelated test that also happens to render the plugin.

Another approach, maybe even a better one, would be figuring out how to best avoid having a truly global globalAppRegistry in tests at all. Ideally, tests should probably use a distinct one each time, not share the same on across all tests in that process…?

Reply:
Ah, I see what you mean. Then I think we can change the code there that passes `globalAppRegistry` value to the context to be `_globalAppRegistry ?? new AppRegistry()`: if you don't care about the reigstry or don't need to interact with it inside the actual test, this would be just a safe fallback, but if you want to emit events or something, then you'd want to pass your instance there. How does that sound?

Reply:
Yeah, I think that makes for some nice isolation between tests, and it should be easy enough to work around if it we do want the actual global instance?

Reply:
Totally, and it would be super explicit if you want to pass the global one, it's a singleton export from the app-registry package. Will update, thanks for the feedback!

Reply:
Updated `globalAppRegistry` to be always scoped as discussed in https://github.com/mongodb-js/compass/pull/5100/commits/72c9c1587fe316133f4b3c43e1c20e5665dbd946. Also when doing that stumled on something that I think is a bug: we did call `plugin.deactivate` in `appRegistry.deactivate`, but didn't de-register plugins, changed that too to always clean-up plugins from registry as at the moment we call deactivate there, we expect this whole thing to be disposeable basically